Makefile

```diff
@@ -70,7 +70,7 @@ verify:
 	go run ./hack/validate-imports/validate-imports.go cmd hack pkg test
 
 unit: generate
-	go test ./... -coverprofile=coverage.out -covermode=atomic
+	go test ./... -coverprofile=coverage.out -covermode=atomic -race -timeout 20m
```

Comment:
A 20m timeout is really long for unit tests! Perhaps we should only include the race detection in long-running CI/e2e tests.

Reply:
Yes I agree :) The tests in `pkg/plugin` are just monsters and take more time with `-race` enabled and even more time when all of that is run on those highly-cpu-constrained containers in CI.

[This commit](https://github.com/openshift/openshift-azure/pull/1687/commits/df8fc5085bd64a14dfbbf103eaa174c531d87160) already improves the situation significantly notably when running with multiple cores. While this optimization will probably not make much of a difference in CI, I imagine that such optimizations could increase the speed of our sync reconciliation on clusters and could lead to standing up a healthy cluster in less time.